payjoin/src/uri.rs

```diff
@@ -274,45 +309,159 @@ enum InternalPjParseError {
     UnsecureEndpoint,
 }
 
+impl<'a> bip21::SerializeParams for &'a Payjoin {
+    type Key = &'static str;
+    type Value = String;
+    type Iterator = std::vec::IntoIter<(Self::Key, Self::Value)>;
+
+    fn serialize_params(self) -> Self::Iterator {
+        match self {
+            Payjoin::Supported(params) => params.serialize_params(),
+            Payjoin::V2Only(params) => params.serialize_params(),
+        }
+    }
+}
+
+impl<'a> bip21::SerializeParams for &'a PayjoinParams {
+    type Key = &'static str;
+    type Value = String;
+    type Iterator = std::vec::IntoIter<(Self::Key, Self::Value)>;
+
+    fn serialize_params(self) -> Self::Iterator {
+        vec![
+            ("pj", self.endpoint.as_str().to_string()),
+            ("pjos", if self.disable_output_substitution { "1" } else { "0" }.to_string()),
+            #[cfg(feature = "v2")]
+            (
+                "ohttp",
+                if let Some(config) = &self.ohttp_config {
+                    bitcoin::base64::encode_config(
+                        config.encode().expect("ohttp config serialization failed"),
```

Comment:
the message here is just a placeholder, as iam not sure how to write a useful message here. 
the encoding could fail only if writing to the buffer fails and i am not sure how verbose/technical to be here..

Reply:
So close!

I think something like this could work

```rs
    fn serialize_params(self) -> Self::Iterator {
        let mut params = vec![
            ("pj", self.endpoint.as_str().to_string()),
            ("pjos", if self.disable_output_substitution { "1" } else { "0" }.to_string()),
        ];
        #[cfg(feature = "v2")]
        if let Ok(config) = &self.ohttp_config.encode() {
            let encoded_config = bitcoin::base64::encode_config(config, bitcoin::base64::URL_SAFE);
            params.push(("ohttp", encoded_config));
        } else {
            log::warn!("Failed to encode ohttp config, ignoring");
        }
        params.into_iter()
    }
```